ouroboros-network-framework/test/Test/Ouroboros/Network/Server2.hs

```diff
@@ -2075,6 +2133,36 @@ prop_connection_manager_valid_transitions serverAcc (ArbDataFlow dataFlow)
                        (Script (toBearerInfo absBi :| [noAttenuation]))
                        maxAcceptedConnectionsLimit events
 
+-- | Property wrapping `multinodeExperiment`.
+--
+-- Note: this test coverage of connection manager state transitions.
+-- TODO: Fix transitions that are not covered. #3516
+prop_connection_manager_transitions_coverage :: Int
+                                             -> ArbDataFlow
+                                             -> AbsBearerInfo
+                                             -> MultiNodeScript Int TestAddr
+                                             -> Property
+prop_connection_manager_transitions_coverage serverAcc (ArbDataFlow dataFlow)
+                                             absBi
+                                             (MultiNodeScript
+                                                events) =
+  let trace = runSimTrace sim
+
+      abstractTransitionEvents :: [AbstractTransitionTrace SimAddr]
+      abstractTransitionEvents = withNameTraceEvents trace
+
+      transitionsSeen = nub [ tran | TransitionTrace _ tran <- abstractTransitionEvents]
+      transitionsSeenNames = map (snd . validTransitionMap) transitionsSeen
+
+   in coverTable "valid transitions" [ (n, 0.01) | n <- allValidTransitionsNames ] $
```

Comment:
does the property passes if we use `checkCoverage`?

Reply:
It won't because there are transitions that are not covered due to the way the generator is generated

Reply:
Have you tried to improve the generator? If so, what about filtering the offending ones?

Reply:
These are examples of transitions that are not covered:

```
Table 'valid transitions' had only 0.00% DuplexSt → InboundSt Duplex, but expected 0.01%
Table 'valid transitions' had only 0.00% DuplexSt → TerminatedSt, but expected 0.01%
Table 'valid transitions' had only 0.00% DuplexSt → TerminatingSt, but expected 0.01%
Table 'valid transitions' had only 0.00% InboundIdleSt Duplex → InboundIdleSt Duplex, but expected 0.01%
Table 'valid transitions' had only 0.00% InboundSt Duplex → TerminatingSt, but expected 0.01%
Table 'valid transitions' had only 0.00% InboundSt Unidirectional → TerminatingSt, but expected 0.01%
Table 'valid transitions' had only 0.00% OutboundDupSt Expired → OutboundDupSt Expired, but expected 0.01%
Table 'valid transitions' had only 0.00% OutboundDupSt Expired → TerminatingSt, but expected 0.01%
Table 'valid transitions' had only 0.00% OutboundDupSt Ticking → OutboundDupSt Ticking, but expected 0.01%
Table 'valid transitions' had only 0.00% OutboundDupSt Ticking → TerminatingSt, but expected 0.01%
Table 'valid transitions' had only 0.00% OutboundIdleSt Duplex → OutboundIdleSt Duplex, but expected 0.01%
Table 'valid transitions' had only 0.00% OutboundIdleSt Unidirectional → InboundSt Unidirectional, but expected 0.01%
Table 'valid transitions' had only 0.00% OutboundIdleSt Unidirectional → OutboundIdleSt Unidirectional, but expected 0.01%
Table 'valid transitions' had only 0.00% OutboundUniSt → TerminatingSt, but expected 0.01%
Table 'valid transitions' had only 0.00% ReservedOutboundSt → TerminatingSt, but expected 0.01%
Table 'valid transitions' had only 0.00% TerminatedSt → ReservedOutboundSt, but expected 0.01%
Table 'valid transitions' had only 0.00% TerminatedSt → UnnegotiatedSt Inbound, but expected 0.01%
Table 'valid transitions' had only 0.00% UnknownConnectionSt → TerminatedSt, but expected 0.01%
Table 'valid transitions' had only 0.00% UnknownConnectionSt → TerminatingSt, but expected 0.01%
Table 'valid transitions' had only 0.00% UnnegotiatedSt Inbound → TerminatingSt, but expected 0.01%
Table 'valid transitions' had only 0.00% UnnegotiatedSt Outbound → TerminatingSt, but expected 0.01%
Table 'valid transitions' had only 0.00% WaitRemoteIdleSt → TerminatedSt, but expected 0.01%
Table 'valid transitions' had only 0.00% WaitRemoteIdleSt → TerminatingSt, but expected 0.01%
```

There are some here that I do not think filtering would make sense. I'll try to see if tweaking the generator a little bit makes so that some of these start being covered.

Reply:
* `DuplexSt → InboundSt Duplex` - nice to cover
* `OutboundIdleSt Unidirectional → InboundSt Unidirectional` - shouldn't it be invalid?
I think all the others could be ignored.

Reply:
> OutboundIdleSt Unidirectional → InboundSt Unidirectional - shouldn't it be invalid?

Yes, good catch!

Reply:
Addressed